# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/FetchChild.cpp
# Commit: 84162d09eef8
# Full Hash: 84162d09eef858c42992fa6cc623284bdab1fd05
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-12 21:30:33
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - PFetch protocol declaration and implementation. r=dom-worker-reviewers,jesup
#   
#   Depends on D138813
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142436
# ==============================================================================

diff -r 5c3b5a384f2f -r 84162d09eef8 dom/fetch/FetchChild.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/fetch/FetchChild.cpp	Thu Jan 12 15:09:45 2023 +0000
@@ -0,0 +1,43 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "FetchChild.h"
+
+namespace mozilla::dom {
+
+mozilla::ipc::IPCResult FetchChild::Recv__delete__(const nsresult&& aResult) {
+  return IPC_OK();
+}
+
+mozilla::ipc::IPCResult FetchChild::RecvOnResponseAvailableInternal(
+    ParentToChildInternalResponse&& aResponse) {
+  // TODO: Should perform WorkerFetchResolver::OnResponseAvailableInternal here.
+  return IPC_OK();
+}
+
+mozilla::ipc::IPCResult FetchChild::RecvOnResponseEnd(ResponseEndArgs&& aArgs) {
+  // TODO: Should perform WorkerFetchResolver::OnResponseEnd here.
+  return IPC_OK();
+}
+
+mozilla::ipc::IPCResult FetchChild::RecvOnDataAvailable() {
+  // TODO: Should perform WorkerFetchResolver::OnDataAvailable here.
+  return IPC_OK();
+}
+
+mozilla::ipc::IPCResult FetchChild::RecvOnFlushConsoleReport(
+    nsTArray<net::ConsoleReportCollected>&& aReports) {
+  // TODO: Should perform WorkerFetchResolver::FlushConsoleReport here.
+  return IPC_OK();
+}
+
+mozilla::ipc::IPCResult FetchChild::RecvOnCSPViolationEvent(
+    const nsAString& aJSON) {
+  // TODO: Should dispatch csp violation event to worker scope.
+  return IPC_OK();
+}
+
+void FetchChild::ActorDestroy(ActorDestroyReason aReason) {}
+
+}  // namespace mozilla::dom